File: algorithms/LinearSearch.java
//package algorithms;
public class LinearSearch {
    public static void main(String[] args) {
        int [] num = {90,10,23,45,12,67,36,87}; //it can be user intput of from a file  
        
        int key = -9; //any number to search for
        boolean flag = false;
       
        int i=0;
        while(i<num.length){
            if(num[i] == key){
                flag=true;
                break;
            }
            i++;
        }
        
        if(flag){
            System.out.println(key+" found at index: "+ i);
        }else{
            System.out.println(key+" not found");
        }
    }
}

/*TO DO: 
1. Implement Linear Search as a method/function which returns the index.
2. Implement without flag variable.
3. Take user input
4. Take input from a text file
*/

File: algorithms/linear_search.c
#include<stdio.h>

int linear_search(int arr[], int k){
    
    for(int index = 0; index<sizeof(arr); index++){
        if(arr[index]==k){
            return index;
        }
    }
    return -1;
}


int main(){
    
    int numbers[5] = {10, 20, 30, 40 , 50};
    int key= 40;
    int i = linear_search(numbers, key);
    
    if(i==-1){
        printf("not found");
    }else{
        printf("found");
    }
}


